### PR TITLE
Hassan/issue#8

### PR DESCRIPTION
**Pull request by hassanraza164**
# Pull Request Template

## Prequisites

- [X] Did you run lint -cf `filename` on your changes, if you are commiting a .c file?
- [ ] Did you run lint -sf `filename` on your changes, if you are commiting a .sh file?
- [X] Does your code compile properly using `make`?
- [X] Did you related issue number in the description?

## Description

#8 has been resolved

## Technical Debt

<Write about any missing things that you are going to do in the future related to this PR>

### CODE REVIEW

File: code/include/env/mtvec32.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_MTVEC32_H
+#define RISCV_MTVEC32_H
+
+#define MTVEC32_MODE_SHIFT 0
+#define MTVEC32_BASE_SHIFT 2
+
+#define MTVEC32_MODE_MASK 0x00000003ull
```
Comment (mmhus) [inline code/include/env/mtvec32.h]: delete file, this isn't what I meant

File: code/include/env/medeleg.h [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MEDELEG_H
```
Comment (mmhus) [inline code/include/env/medeleg.h:1]: delete file, this isn't what I meant

File: code/include/env/medeleg32.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MEDELEG32_H
```
Comment (mmhus) [inline code/include/env/medeleg32.h]: delete file, this isn't what I meant

File: code/include/env/medeleg64.h [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MEDELEG64_H
```
Comment (mmhus) [inline code/include/env/medeleg64.h]: delete file, this isn't what I meant

File: code/include/env/medeleg_32.h [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MEDELEG_H
```
Comment (mmhus) [inline code/include/env/medeleg_32.h]: delete file, this isn't what I meant

File: code/include/env/mideleg.h [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MIDELEG_H
```
Comment (mmhus) [inline code/include/env/mideleg.h:1]: delete file, this isn't what I meant

File: code/include/env/mideleg32.h [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MIDELEG32_H
```
Comment (mmhus) [inline code/include/env/mideleg32.h]: delete file, this isn't what I meant

File: code/include/env/mideleg64.h [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MIDELEG64_H
```
Comment (mmhus) [inline code/include/env/mideleg64.h]: delete file, this isn't what I meant

File: code/include/env/mideleg_32.h [hunk 9]
```diff
@@ -0,0 +1,8 @@
+#ifndef RISCV_MIDELEG_H
```
Comment (mmhus) [inline code/include/env/mideleg_32.h]: delete file, this isn't what I meant

File: code/include/env/mtvec.h [hunk 10]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_MTVEC_H
```
Comment (mmhus) [inline code/include/env/mtvec.h:1]: Please also add defines for MTVEC  mode field types, e.g. direct and vectored

File: code/include/env/mtvec64.h [hunk 11]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_MTVEC64_H
```
Comment (mmhus) [inline code/include/env/mtvec64.h]: delete file, this isn't what I meant

File: code/include/env/mtvec_32.h [hunk 12]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_MTVEC_H
```
Comment (mmhus) [inline code/include/env/mtvec_32.h]: delete file, this isn't what I meant

File: code/include/env/stvec.h [hunk 13]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_STVEC_H
```
Comment (mmhus) [inline code/include/env/stvec.h:1]: delete file, this isn't what I meant

File: code/include/env/stvec32.h [hunk 14]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_STVEC32_H
```
Comment (mmhus) [inline code/include/env/stvec32.h]: delete file, this isn't what I meant

File: code/include/env/stvec64.h [hunk 15]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_STVEC64_H
```
Comment (mmhus) [inline code/include/env/stvec64.h]: delete file, this isn't what I meant

File: code/include/env/stvec_32.h [hunk 16]
```diff
@@ -0,0 +1,10 @@
+#ifndef RISCV_STVEC_H
```
Comment (mmhus) [inline code/include/env/stvec_32.h]: delete file, this isn't what I meant

File: code/include/env/medeleg.h (hunk 2 above)
Comment (mmhus) [reply code/include/env/medeleg.h:1]: I asked you take a look at mcause. It seems you haven't understood the purpose of this register. We will leave it for now. Try and understand the purpose of these registers, in addition also look at mcause and see what it does first.

**Review by mmhus — CHANGES_REQUESTED** [review]

File: code/include/env/mtvec.h (hunk 10 above)
Comment (hassanraza164) [reply code/include/env/mtvec.h:1]: Values for direct and vectored has been added

File: code/include/env/medeleg.h (hunk 2 above)
Comment (hassanraza164) [reply code/include/env/medeleg.h:1]: i have modified the file again

File: code/include/env/mideleg.h (hunk 6 above)
Comment (hassanraza164) [reply code/include/env/mideleg.h:1]: File has been changed

File: code/include/env/stvec.h (hunk 13 above)
Comment (hassanraza164) [reply code/include/env/stvec.h:1]: File has been modified

**Review by mmhus — APPROVED** [review]

**Pull request merged**